### PR TITLE
Code coverage integration

### PR DESCRIPTION
**Pull request by nicholashusin**
This PR integrates the statement code coverage feature with the entire web application.

Screenshot:
- [Code coverage on solutions page](https://screenshot.googleplex.com/TYzvMyDkdKt.png)
- [Code coverage on profile page](https://screenshot.googleplex.com/K8o3pFXBqOp.png)

### CODE REVIEW

File: routes/problems_id.js [hunk 1]
```diff
@@ -26,11 +27,24 @@ async function getProblem(id) {
...
 
+/**
+ * Helper function that calculates code coverage.
+ * Multiplies the timeout given by CODE_COVERAGE_TIMEOUT_MULTIPLIER
+ * to account for instrumented code taking longer to run
+ * and returns it as a datastore double.
+ **/
+async function calculateCodeCoverage(code, timeout) {
+  const coverageTimeout = timeout * CODE_COVERAGE_TIMEOUT_MULTIPLIER;
+  const codeCoverage = await sandbox.calculateCodeCoverage(code);
```
Comment (ifeomi) [inline routes/problems_id.js]: it doesn't look like the timeout is used, did u mean to pass it as a param to `calculateCodeCoverage`

Comment (nicholashusin) [reply routes/problems_id.js]: Thanks for the catch!
Fixed it :)

File: routes/problems_id.js [hunk 2]
```diff
@@ -40,6 +54,8 @@ async function saveSubmission(user, code, analysisResult, problemId) {
   data.username     = user.username;
   data.code         = code;
   data.problemId    = parseInt(problemId);
+  data.coverage     = coverage;
+  console.log(data);
```
Comment (ifeomi) [inline routes/problems_id.js]: leftover from debugging?

**Review by ifeomi — APPROVED** [review]
lgtm!

**Review by lalaAdasia00 — APPROVED** [review]

**Review by chronologos — APPROVED** [review]

File: routes/problems_id.js (hunk 2 above)
Comment (nicholashusin) [reply routes/problems_id.js]: Thanks for the catch, fixed now :)

**Pull request merged**